activerecord/lib/active_record/validations/uniqueness.rb

```diff
@@ -29,7 +29,22 @@ def validate_each(record, attribute, value)
           end
         end
         relation = scope_relation(record, relation)
-        relation = relation.merge(options[:conditions]) if options[:conditions]
+
+        if options[:conditions]
+          conditions = options[:conditions]
+
+          relation = if conditions.arity.zero?
+            relation.merge(conditions)
+          else
+            # Support passing record to build conditions dynamically using attributes of record.
+            # E.g. validate slug must be unique for the day of publication:
+            # validates_uniqueness_of :slug,
+            #   conditions: ->(record) {
+            #     where(published_at: record.published_at.beginning_of_day..record.published_at.end_of_day)
+            #   }
```

Comment:
This example should be incorporated into the API documentation instead, probably [here](https://github.com/rails/rails/blob/ae5ecfe26c8fdc194d5b3cd16fd13b64574bcfd3/activerecord/lib/active_record/validations/uniqueness.rb#L128).